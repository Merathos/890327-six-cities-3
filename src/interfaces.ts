export interface City {
  name: string;
  coords: [number, number];
  zoom: number;
};

export interface Offer {
  id: string;
  city: City;
  coords: [number, number];
  zoom: number;
  title: string;
  previewImg: string;
  photos: string[];
  bedroomsAmount: number;
  maxAdults: number;
  features: string[];
  type: string;
  rating: number;
  isBookmarked: boolean;
  isPremium: boolean;
  price: number;
  hostName: string;
  hostAvatar: string;
  hostStatus: boolean;
  description: string;
};

export interface User {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface Comment {
  id: string;
  name: string;
  avatar: string;
  rating: number;
  date: string;
  text: string;
}
