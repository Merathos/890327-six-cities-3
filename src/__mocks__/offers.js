export default [
  {
    id: `41324`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.85309666406198],
    zoom: 10,
    title: `Beautiful & luxurious apartment at great location`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.369553943508, 4.85309666406198],
    title: `Wood and stone place`,
    previewImg: `img/apartment-02.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    previewImg: `img/apartment-03.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    city: {
      name: `Paris`,
      coords: [53.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3809553943508, 4.939309666406198],
    title: `Nice, cozy, warm big bed apartment`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
