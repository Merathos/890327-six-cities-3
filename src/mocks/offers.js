export default [
  {
    id: `41324`,
    name: `Beautiful &amp; luxurious apartment at great location`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    name: `Wood and stone place`,
    picture: `apartment-02`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    name: `Canal View Prinsengracht`,
    picture: `apartment-03`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    name: `Nice, cozy, warm big bed apartment`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
