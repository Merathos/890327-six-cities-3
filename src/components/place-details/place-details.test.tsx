import * as React from "react";
import * as renderer from "react-test-renderer";
import PlaceDetails from "./place-details";
import offers from "../../__mocks__/offers";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import comments from "../../__mocks__/reviews";
import {BrowserRouter as Router} from "react-router-dom";
import {Offer, City} from "../../interfaces";

const mockStore = configureStore([]);

const cities: City[] = [
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
];

const hoveredCard: Offer = offers[0];

it(`Render place-details`, () => {
  const store = mockStore({
    DATA: {
      offers,
      cities,
      currentCity: {
        name: `Amsterdam`,
        coords: [52.370216, 4.895168],
        zoom: 10
      },
      detailsOfferID: `41324`,
      nearbyOffers: offers,
      comments,
      bookmarkStatus: ``

    },
    APPLICATION: {
      currentSortType: `Popular`,
      hoveredCard
    },
    USER: {
      isAuthorized: false,
      user: {}
    }
  });

  store.dispatch = jest.fn();
  const tree = renderer
    .create(
        <Router>
          <Provider store={store}>
            <PlaceDetails match={{params: {id: `41324`}, isExact: true, path: ``, url: ``}} />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
