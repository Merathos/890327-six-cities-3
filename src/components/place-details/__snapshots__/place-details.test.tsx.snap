// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render place-details 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1 bedroom
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 2 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                4
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "50%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Consectetur sit velit nulla ipsum. Sunt occaecat amet cupidatat laboris commodo in. Culpa voluptate elit amet officia veniam cillum ad. Ea magna adipisicing laboris incididunt. Laborum eu sit consequat eu nisi. Nulla ipsum ea reprehenderit enim labore do et. Irure anim laborum aute quis.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-06-13"
                  >
                    June 2019
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-angelina.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Angelina
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "72%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Amet anim Lorem laborum ullamco nisi labore adipisicing ut esse eu. Consectetur laboris consectetur labore ut aute do. Eiusmod sit anim adipisicing excepteur reprehenderit in. Aliqua id laborum nisi veniam dolore cillum esse Lorem non occaecat do mollit.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2018-05-13"
                  >
                    May 2018
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "94%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Aliqua deserunt quis eu culpa nulla. Dolor eiusmod aliquip nulla commodo excepteur ut labore Lorem duis consequat dolor ex aute cillum. Non nisi do eu ex mollit sunt quis cillum fugiat velit et. Aliqua magna excepteur minim adipisicing aute consectetur exercitation laboris laborum sint pariatur in esse consectetur. Elit sit incididunt pariatur est eiusmod non anim deserunt laboris incididunt minim reprehenderit.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2017-04-13"
                  >
                    April 2017
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-angelina.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Angelina
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "116%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Velit ipsum voluptate consectetur eiusmod excepteur aliqua et cillum laboris anim nisi dolor. Ex esse id occaecat ex consequat duis sint duis excepteur. Fugiat est nisi dolor officia nisi id. Irure ullamco do ea eiusmod fugiat enim fugiat laborum ea ipsum cupidatat consectetur. Consectetur consequat esse velit tempor commodo ea labore.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2016-03-13"
                  >
                    March 2016
                  </time>
                </div>
              </li>
            </ul>
            
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          id="map"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    120
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active "
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/41324"
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment at great location
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-02.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button "
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/73645"
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private room
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-03.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    132
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active "
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/23455"
                  onClick={[Function]}
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>,
]
`;
